peg_src/jsdoctype.pegjs

```diff
@@ -28,11 +28,11 @@ TopTypeExpr = _ expr:( VariadicTypeExpr
                   / ArrayTypeExpr
                   / GenericTypeExpr
                   / RecordTypeExpr
-                  / TupleTypeExpr
-                  / ArrowTypeExpr
-                  / FunctionTypeExpr
-                  / TypeQueryExpr
-                  / KeyQueryExpr
+                  / TupleTypeExpr // no-jsdoc, no-closure
+                  / ArrowTypeExpr // no-jsdoc, no-closure
+                  / FunctionTypeExpr // no-jsdoc
```

Comment:
does jsdoc mean jsdoc as described by jsdoc.app? 3.2 and above [say that they support closure syntax](https://jsdoc.app/tags-type.html), which I thought meant `function()` types.

Reply:
Ugh... I somehow missed that and yeah was going by the explicit docs. Their lack of prior documentation may indicate a difference from a prior version as it looks like Closure syntax wasn't fully supported before, but I don't see value in distinguishing this now. However, the jsdoc docs also mention that jsdoc supports other types not supported by Closure ("several other formats that are specific to JSDoc."); it seems the `<type>[]` format at least (e.g., `string[]`) is not supported in Closure per jsdoc (and I don't see it on the Closure type pages). Guess need to do more homework here...

Reply:
I've just now pushed a commit to prohibit this as jsdoc's `catharsis` also seems to think Closure doesn't support. However, to be sure, I've asked at https://stackoverflow.com/questions/60578125/does-closure-support-jsdoc-array-syntax-like-string (and also asked about optional keys in Record types). (Download speeds are poor here, so can't search too easily through Closure's repo at the moment.)